packages/env/lib/download-sources.js

```diff
@@ -85,6 +86,8 @@ async function downloadSource( source, options ) {
 	} else if ( source.type === 'zip' ) {
 		await downloadZipSource( source, options );
 	}
+
+	await fs.promises.chmod( source.path, 0o777 );
```

Comment:
This seems it would only be applying to the path and it wouldn't be recursively applying the to all files in the directory tree, right?

Reply:
The [`chmodr`](https://www.npmjs.com/package/chmodr) package should be used instead, perhaps. However, I can't vouch for it.

Might be better to use the approach in https://github.com/WordPress/gutenberg/issues/22515#issuecomment-1119094610.